openmc/data/photon.py

```diff
@@ -929,83 +926,6 @@ def _add_bremsstrahlung(self):
         self.bremsstrahlung.update(_BREMSSTRAHLUNG[self.atomic_number])
 
-    def _compute_heating(self):
-        r"""Compute heating cross sections (KERMA)
-
-        Photon energy is deposited as energy loss in three reactions:
-        incoherent scattering, pair production and photoelectric effect.
-        The point-wise heating cross section is calculated as:
-
-        .. math::
-            \begin{aligned}
-            \sigma_{Hx}(E) &= (E - \overline{E}_x(E)) \cdot \sigma_x(E), x \in \left\{I, PP, PE \right\} \\
-            \overline{E}_I(E) &= \frac {\int E' \sigma_I (E,E',\mu) d\mu} {\int \sigma_I (E,E',\mu) d\mu} \\
-            \overline{E}_{PP} &= 2 m_e c^2 = 1.022 \times 10^6 eV \\
-            \overline{E}_{PE} &= E(\text{fluorescent photons})
-            \end{aligned}
-
-        The differential cross section representation for incoherent
-        scattering can be found in the theory manual.
-
-        """
-
-        # Determine a union energy grid
-        energy = np.array([])
-        for mt in (504, 515, 517, 522):
-            if mt in self:
-                energy = np.union1d(energy, self[mt].xs.x)
-
-        heating_xs = np.zeros_like(energy)
-
-        # Incoherent scattering
-        if 504 in self:
-            rx = self[504]
-
-            def dsigma_dmu(mu, E):
-                k = E / MASS_ELECTRON_EV
-                krat = 1.0 / (1.0 + k * (1.0 - mu))
-                x = E * sqrt(0.5 * (1.0 - mu)) / PLANCK_C
-                return pi * R0*R0 * krat*krat * (krat + 1/krat +
-                       mu*mu - 1.0) * rx.scattering_factor(x)
-
-            def eout_dsigma_dmu(mu, E):
-                Eout = E / (1.0 + E / MASS_ELECTRON_EV * (1.0 - mu))
-                return Eout * dsigma_dmu(mu, E)
-
-            def eout_average(E):
-                integral_sigma = quad(dsigma_dmu, -1.0, 1.0,
-                                      args=(E,), epsabs=0.0, epsrel=1e-3)[0]
-                integral_sigma_e = quad(eout_dsigma_dmu, -1.0, 1.0,
-                                        args=(E,), epsabs=0.0, epsrel=1e-3)[0]
-                return integral_sigma_e / integral_sigma
-
-            e_out = np.vectorize(eout_average)(energy)
-            heating_xs += (energy - e_out) * rx.xs(energy)
-
-        # Pair production, electron field
-        if 515 in self:
-            heating_xs += (energy - 2*MASS_ELECTRON_EV)*self[515].xs(energy)
-
-        # Pair production, nuclear field
-        if 517 in self:
-            heating_xs += (energy - 2*MASS_ELECTRON_EV)*self[517].xs(energy)
-
-        # Photoelectric effect
-        if 522 in self:
-            # Account for fluorescent photons
-            for mt, rx in self.reactions.items():
-                if mt >= 534 and mt <= 572:
-                    shell = _REACTION_NAME[mt][1]
-                    relax_data = self.atomic_relaxation
-                    if relax_data is not None:
-                        e_f = relax_data.energy_fluorescence(shell)
```

Comment:
Is there still a situation where we might want to calculate the expected energy of the fluorescence photons now that we’re not storing the heating numbers (i.e., is there a reason to keep the method `energy_fluorescence`)?

Reply:
Good point. That method is no longer used anywhere, so I'll go ahead and remove it. If we ever want/need to revive it, it's still there in the git history :)

Reply:
:heavy_check_mark: